eos-tech-support/eos-diagnostics

```diff
@@ -196,180 +196,127 @@ function tryReadFile(filename, fallbackMessage) {
-    fullDump += trySpawn('lspci -knn');
-    fullDump += '\n';
-    fullDump += 'USB\n\n';
-    fullDump += trySpawn('lsusb');
-    fullDump += '\n';
-
-    fullDump += '========================\n'
-    fullDump += '= Printers information =\n'
-    fullDump += '========================\n'
-    fullDump += '\n';
-    fullDump += collectPrintersInfo();
-    fullDump += '\n';
-
-    fullDump += '===========================\n'
-    fullDump += '= Temperature information =\n'
-    fullDump += '===========================\n'
-    fullDump += '\n';
-    fullDump += collectTemperatureInfo();
-    fullDump += '\n';
-
-    fullDump += '================================\n'
-    fullDump += '= Chromium plugins information =\n'
-    fullDump += '================================\n'
-    fullDump += '\n';
-    fullDump += collectChromiumPluginsInfo();
-    fullDump += '\n';
-
-    fullDump += '======================\n'
-    fullDump += '= Codecs information =\n'
-    fullDump += '======================\n'
-    fullDump += '\n';
-    fullDump += trySpawn('find /var/lib/codecs');
-    fullDump += '\n';
-    fullDump += trySpawn('gst-inspect-1.0 libav');
-    fullDump += trySpawn('gst-inspect-1.0 -b');
-    fullDump += '\n';
-
-    fullDump += '================================\n'
-    fullDump += '= Flatpak remote configuration =\n'
-    fullDump += '================================\n'
-    fullDump += '\n';
-    fullDump += trySpawn('flatpak remote-list --show-details');
-    fullDump += '\n';
-
-    fullDump += '==============================\n'
-    fullDump += '= Flatpak installed runtimes =\n'
-    fullDump += '==============================\n'
-    fullDump += '\n';
-    fullDump += trySpawn('flatpak list --runtime --show-details');
-    fullDump += '\n';
-
-    fullDump += '==================================\n'
-    fullDump += '= Flatpak installed applications =\n'
-    fullDump += '==================================\n'
-    fullDump += '\n';
-    fullDump += trySpawn('flatpak list --app --show-details');
-    fullDump += '\n';
-
-    fullDump += '================\n'
-    fullDump += '= Process list =\n'
-    fullDump += '================\n'
-    fullDump += '\n';
-    fullDump += trySpawn('ps aux');
-    fullDump += '\n';
-
-    fullDump += '===============\n'
-    fullDump += '= Boot timing =\n'
-    fullDump += '===============\n'
-    fullDump += '\n';
-    fullDump += trySpawn('systemd-analyze');
-    fullDump += '\n';
-
-    fullDump += '==================\n';
-    fullDump += '= Journal output =\n';
-    fullDump += '==================\n';
-    fullDump += '\n';
-    fullDump += trySpawn('journalctl --no-pager');
+    for (var i = 0; i < diagnostics.length; i++) {
+        let diag = diagnostics[i];
+        fullDump += "=".repeat(diag['title'].length + 4) + '\n';
+        fullDump += "= " + diag['title'] + ' =\n';
+        fullDump += "=".repeat(diag['title'].length + 4) + '\n';
+        let content = diag['content']().toString().trim();
```

Comment:
The calls to `toString()` and `trim()` do not seem required here, since we did not have them before. Any reason you introduced them?